pkg/healthcheck/healthcheck.go

```diff
@@ -1440,6 +1440,9 @@ func CheckProxyVersionsUpToDate(pods []corev1.Pod, versions version.Channels) er
 		}
 	}
 	if len(outdatedPods) > 0 {
+		if len(versions.Array) == 0 {
```

Comment:
what do you think of moving this up to before we loop over the pods?  If the versions channels is empty, we already know that none of the pods will match.